sklearn/cluster/_k_means_elkan.pyx

```diff
@@ -0,0 +1,140 @@
+# cython: cdivision=True
+# cython: boundscheck=False
+# cython: wraparound=False
+#
+# Author: Andreas Mueller
+#
+# Licence: BSD 3 clause
+
+import numpy as np
+cimport numpy as np
+cimport cython
+
+from libc.math cimport sqrt
+
+from ..metrics import euclidean_distances
+from ._k_means import _centers_dense
+
+
+cdef double *getfloatpointer(np.ndarray[np.float64_t, ndim=2, mode='c'] data):
+    return <double *>(data.data) 
+
+
+cdef double d(double* a, double* b, int n_features) nogil:
+    cdef double result, tmp
+    result = 0
+    cdef int i
+    for i in range(n_features):
+        tmp = (a[0] - b[0])
+        a += 1
+        b += 1
+        result += tmp * tmp
+    return sqrt(result)
+
+cdef assign_labels(double* X, double* centers, double[:, :]
+                   center_distances, int[:] labels, double[:, :] lower_bounds, double[:] distances, int n_samples, int n_features, int n_clusters):
```

Comment:
well, it updates labels and distances. how would you call it? ``update_labels_distances``?